src/payment/strategies/googlepay/googlepay-payment-processor.ts

```diff
@@ -156,11 +156,12 @@ export default class GooglePayPaymentProcessor {
     }
 
     private _mapGooglePayAddressToBillingAddress(paymentData: GooglePaymentData, id: string): BillingAddressUpdateRequestBody {
-        const firstName = paymentData.paymentMethodData.info.billingAddress.name.split(' ').slice(0, -1).join(' ');
+        const name = paymentData.paymentMethodData.info.billingAddress.name.split(' ');
```

Comment:
People won't expect an array if we use a singular noun.

It's also a bit odd to split the names here.

Given the array isn't used anywhere else in this scope, it should probably happen within `_getFirstAndLastName`.

I'd recommend losing the split (putting that in the helper function) and moving the use of `_getFirstAndLastName` right next to it:

```
        const fullName = paymentData.paymentMethodData.info.billingAddress.name;
        const [firstName, lastName] = this._getFirstAndLastName(fullName);
```